### PR TITLE
Add color balance(Hue, Saturation, Brightness, Contrast) adjustment o…

### PR DESCRIPTION
**Pull request by feiwan1**
…n VA path.

Jira: None.
Test: Build passes on Linux.

### CODE REVIEW

**Comment by kalyankondapally** [comment]
@XiaosongWei  @xuguangxin Can you please review

File: common/compositor/va/varenderer.cpp [hunk 1]
```diff
@@ -199,6 +199,80 @@ bool VARenderer::Init(int gpu_fd) {
   return ret == VA_STATUS_SUCCESS ? true : false;
```
Comment (kalyankondapally) [inline common/compositor/va/varenderer.cpp]: Please update commit message correctly

File: common/compositor/va/varenderer.cpp [hunk 2]
```diff
@@ -294,6 +367,67 @@ bool VARenderer::Draw(const MediaState& state, NativeSurface* surface) {
   param.filters = nullptr;
   param.filter_flags = VA_FRAME_PICTURE;
 
+  if(caps[0].type == VAProcColorBalanceNone) {
```
Comment (XiaosongWei) [inline common/compositor/va/varenderer.cpp]: please disable this feature by default until we have related interface to pass the settings from app to hwc.

File: common/compositor/va/varenderer.h [hunk 3]
```diff
@@ -35,12 +37,25 @@ class VARenderer : public Renderer {
...
+  bool SetVAProcFilterDefaultValue(VAProcFilterCapColorBalance* caps);
+  float GetVAProcFilterDefaultValue(VAProcColorBalanceType type, VAProcFilterCapColorBalance* caps);
+  bool MapVAProFilterValuetoDriver(VAProcColorBalanceType type, float value);
 
  private:
   int DrmFormatToVAFormat(int format);
   int DrmFormatToRTFormat(int format);
 
   void *va_display_ = nullptr;
+  VAProcFilterCapColorBalance caps[VAProcColorBalanceCount];
```
Comment (XiaosongWei) [inline common/compositor/va/varenderer.h]: coding style

File: common/compositor/va/varenderer.h [hunk 4]
```diff
@@ -35,12 +37,25 @@ class VARenderer : public Renderer {
...
+  uint32_t cap_num = VAProcColorBalanceCount;
+
+  float hue_value=0;
+  float saturation_value=0;
+  float brightness_value=0;
+  float contrast_value=0;
```
Comment (XiaosongWei) [inline common/compositor/va/varenderer.h]: pls add trailing _ for member variables.

File: common/compositor/va/varenderer.cpp (hunk 1 above)
Comment (feiwan1) [reply common/compositor/va/varenderer.cpp]: @kalyankondapally, updated, please have a review.

File: common/compositor/va/varenderer.cpp (hunk 2 above)
Comment (xuguangxin) [inline common/compositor/va/varenderer.cpp]: It's no need support VAProcColorBalanceNone

File: common/compositor/va/varenderer.cpp [hunk 5]
```diff
@@ -294,6 +367,67 @@ bool VARenderer::Draw(const MediaState& state, NativeSurface* surface) {
...
+  }
+
+  ScopedVABufferID filter_hue(va_display_);
+  ScopedVABufferID filter_sturation(va_display_);
+  ScopedVABufferID filter_brightness(va_display_);
+  ScopedVABufferID filter_contrast(va_display_);
+  std::vector<VABufferID> filters;
+  VAProcFilterParameterBufferColorBalance cbparam;
+
+ SetVAProcFilterValue(VAProcColorBalanceBrightness, 100);
```
Comment (xuguangxin) [inline common/compositor/va/varenderer.cpp]: why we need set this to 100

File: common/compositor/va/varenderer.cpp [hunk 6]
```diff
@@ -294,6 +367,67 @@ bool VARenderer::Draw(const MediaState& state, NativeSurface* surface) {
...
+
+  cbparam.type = VAProcFilterColorBalance;
+
+  if(hue_value != GetVAProcFilterDefaultValue(VAProcColorBalanceHue, &caps[0]) ) {
```
Comment (xuguangxin) [inline common/compositor/va/varenderer.cpp]: You can't compare two float values.  http://www.geeksforgeeks.org/comparison-float-value-c/. Can we change it to int?

**Comment by feiwan1** [comment]
Drop this patch, I will ask another pull request for this feature.

**Pull request closed without merging**